test/functional/hooks.js

```diff
@@ -41,38 +43,120 @@ describe('library hooks', () => {
-      var el = container.childNodes[0];
 
-      expect(attributes.applyAttr).calledWith(el, 'staticName', 'staticValue');
+    afterEach(() => {
+      for (var mutator in mutators.attributes) {
+        if (mutator !== '__all') {
+          mutators.attributes[mutator] = null;
+        }
+      }
     });
 
-    it('should be called for dynamic attributes', () => {
-      patch(container, render, 'dynamicValue');
-      var el = container.childNodes[0];
 
-      expect(attributes.applyAttr).calledWith(el, 'dynamicName', 'dynamicValue');
+    describe('for static attributes', () => {
+      it('should call specific setter', () => {
+        stubOut('staticName');
+
+        patch(container, render, 'dynamicValue');
+        var el = container.childNodes[0];
+
+        expect(stub).to.have.been.calledOnce;
+        expect(stub).to.have.been.calledWith(el, 'staticName', 'staticValue');
+      });
+
+      it('should call generic setter', () => {
+        patch(container, render, 'dynamicValue');
+        var el = container.childNodes[0];
+
+        expect(allSpy).to.have.been.calledWith(el, 'staticName', 'staticValue');
+      });
+
+      it('should prioritize specific setter over generic', () => {
+        stubOut('staticName');
+
+        patch(container, render, 'dynamicValue');
+        var el = container.childNodes[0];
+
+        expect(stub).to.have.been.calledOnce;
+        expect(allSpy).to.have.been.calledOnce;
+        expect(allSpy).to.have.been.calledWith(el, 'dynamicName', 'dynamicValue');
+      });
     });
 
-    it('should be called on attribute update', () => {
-      patch(container, render, 'dynamicValueOne');
-      patch(container, render, 'dynamicValueTwo');
-      var el = container.childNodes[0];
+    describe('for specific dynamic attributes', () => {
+      beforeEach(() => {
+        stubOut('dynamicName');
+      });
 
-      expect(attributes.applyAttr).calledWith(el, 'dynamicName', 'dynamicValueTwo');
+      it('should be called for dynamic attributes', () => {
+        patch(container, render, 'dynamicValue');
+        var el = container.childNodes[0];
+
+        expect(stub).to.have.been.calledOnce;
+        expect(stub).to.have.been.calledWith(el, 'dynamicName', 'dynamicValue');
+      });
+
+      it('should be called on attribute update', () => {
+        patch(container, render, 'dynamicValueOne');
+        patch(container, render, 'dynamicValueTwo');
+        var el = container.childNodes[0];
+
+        expect(stub).to.have.been.calledTwice;
+        expect(stub).to.have.been.calledWith(el, 'dynamicName', 'dynamicValueTwo');
+      });
+
+      it('should only be called when attributes change', () => {
+        patch(container, render, 'dynamicValue');
+        patch(container, render, 'dynamicValue');
+        var el = container.childNodes[0];
+
+        expect(stub).to.have.been.calledOnce;
+        expect(stub).to.have.been.calledWith(el, 'dynamicName', 'dynamicValue');
```

Comment:
I would just drop this expect since the calling with correct value on the initial call is covered by another test.